libcalico-go/lib/backend/k8s/resources/ipam_affinity.go

```diff
@@ -372,7 +374,7 @@ func (c *blockAffinityClient) listV1(ctx context.Context, list model.BlockAffini
 		if err != nil {
 			return nil, err
 		}
-		if host == "" || v1kvp.Key.(model.BlockAffinityKey).Host == host {
+		if host == "" || v1kvp.Key.(model.BlockAffinityKey).AffinityCfg.Host == host {
```

Comment:
All of these checks need to also compare the `AffinityCfg.Type` 

for example:

```go
host := v1kvp.Key.(model.BlockAffinityKey).AffinityCfg.Host
affType := v1kvp.Key.(model.BlockAffinityKey).AffinityCfg.Type
if host == "" || (list.Type == affType && host == list.Host) {

Reply:
Without this, we still have the chance of hitting the case where the hostname matches but the affinity type does not match, which would be a problem.